### PR TITLE
Initial Forta Bots

### PR DESCRIPTION
**Pull request by e1Ru1o**
This PR is adding the 2 high priority bots needed for monitoring MakerDAO DAI Bridges & Teleport.

## Bridge Invariant bot
This bot monitors the Bridge Invariant for Optimism & Arbitrum.
#### Considerations:
- Keep in mind the comment that is in the Notion Page with the Monitoring requirements
  > if you teleport DAI from L2A to L2B, you will mint on L2B before you flush and rebalance L1B bridge contract... so during that time you might have an alert  
- More networks can be added if needed.
#### Deployment:
This bot needs a secondary bot that monitors L2. That bot (L2-DAI-Monitor) needs to be deployed first on for Optimism & Arbitrum networks. After that the `BotId` needs to be added in the code of the Bridge-Invariant bot as specified in the README. Once done, the Bridge-Invariant bot can be deployed.

## Teleport Backing bot
This bot detects `Mint` events emitted from the `TeleportJoin` contract (having `TeleportOracleAuth` as the `originator`) for which there doesn't exist a corresponding `WormholeInitialized` event emitted from the `L2DaiWormholeGateway` contract.
#### Considerations:
- Addresses need to be updated once the contracts are deployed. For doing that an entry should be added in the `NETWORK_MAP` variable of `network.ts` with the details of the networks.
- Currently the `network.ts` contains the testnet addresses from the [Wormhole-Integration-Tests](https://github.com/makerdao/wormhole-integration-tests) repo and PoC addresses on Polygon mainnet.
- `L1-Teleport-Backing` bot is expected to run on Ethereum mainnet.
- `WormholeInitialized-Events-Monitoring` bot is expected to run on Optimism & Arbitrum.
- More networks can be added if needed.
#### Deployment:
- As mentioned in the above section, once the Teleport addresses are deployed the bot data should be modified to include those addresses in their respective networks. 
- This bot needs a secondary bot that monitor L2s. That bot (WormholeInitialized-Events-Monitoring) needs to be deployed first on Optimism & Arbitrum networks. After that the `BotId` needs to be added in the code of the L1-Teleport-Backing bot as specified in the README. Once done, the L1-Teleport-Backing bot can be deployed.

### CODE REVIEW

File: forta-bots/Bridge-Invariant/package.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "maker-bridge-invariant",
+  "version": "0.0.1",
+  "description": "Bot to monitor the Maker Bridge Invariant across the L2 networks",
+  "chainIds": [
+    1
+  ],
+  "scripts": {
+    "build": "tsc",
+    "start": "npm run start:dev",
```
Comment (krzkaczor) [inline forta-bots/Bridge-Invariant/package.json:10]: can we use yarn (both for lockfiles and for running scripts)? this is what we use in all other projects in this repo.

File: forta-bots/L1-Teleport-Backing/PoC/TeleportJoin.sol [hunk 2]
```diff
@@ -0,0 +1,288 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
```
Comment (krzkaczor) [inline forta-bots/L1-Teleport-Backing/PoC/TeleportJoin.sol:5]: why sol files are dumped here?

**Review by krzkaczor — CHANGES_REQUESTED** [review]
One major question: how does the `WormholeInitialized-Events-Monitoring` bot record past events if it's called on every new block? Or am I missing something and it will be called with past data too?

File: forta-bots/L1-Teleport-Backing/PoC/TeleportJoin.sol (hunk 2 above)
Comment (e1Ru1o) [reply forta-bots/L1-Teleport-Backing/PoC/TeleportJoin.sol:5]: This contracts are the ones we used as PoC for testing the bots. They have tiny modifications to allow us to easily change some values in the contract so we can test the bots. Each time we create a PoC like this one we include the contracts in the repo, but if you want to, we can delete them.

**Comment by e1Ru1o** [comment]
> One major question: how does the `WormholeInitialized-Events-Monitoring` bot record past events if it's called on every new block? Or am I missing something and it will be called with past data too?

No, you are right. With the current implementation we will be able to detect all the `WormholeInitialized` since the bots deployment, but not the past ones.

File: forta-bots/Bridge-Invariant/package.json (hunk 1 above)
Comment (e1Ru1o) [reply forta-bots/Bridge-Invariant/package.json:10]: I guess it is possible.

Comment (e1Ru1o) [reply forta-bots/Bridge-Invariant/package.json]: changed from npm to yarn :heavy_check_mark:

**Comment by krzkaczor** [comment]
> No, you are right. With the current implementation we will be able to detect all the WormholeInitialized since the bots deployment, but not the past ones.

Are u planning on fixing it? I think this might be a problem in the future during mainnet deployment. We need to deploy bot before any other tx, otherwise, we will get false positives. If you're not planning to fix it please emphasize this point in the readme.

**Comment by e1Ru1o** [comment]
> > No, you are right. With the current implementation we will be able to detect all the WormholeInitialized since the bots deployment, but not the past ones.
> 
> Are u planning on fixing it? I think this might be a problem in the future during mainnet deployment. We need to deploy bot before any other tx, otherwise, we will get false positives. If you're not planning to fix it please emphasize this point in the readme.

fixed

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**